system/Database/BaseConnection.php

```diff
@@ -994,6 +995,11 @@ public function protectIdentifiers($item, bool $prefixSingle = false, ?bool $pro
             return $item;
         }
 
+        // Do not protect identifiers and do not prefix, no swap prefix, there is nothing to do
+        if ($protectIdentifiers === false && $prefixSingle === false && $this->swapPre === '') {
```

Comment:
If I'm not mistaken, if the swapPre property is set, escaping is unavoidable.

Reply:
What do you mean? Is there something wrong with the `if` conditions?

Reply:
If the swapPre property is set (not empty string), the `if` condition will be false.